src/kafka-net/Producer.cs

```diff
@@ -134,14 +136,24 @@ public Producer(IBrokerRouter brokerRouter, int maximumAsyncRequests = MaximumAs
                                 .ToList();
         }
 
+        public Task<List<ProduceResponse>> SendMessageAsync(string topic, int partition, params Message[] messages)
+        {
+            return SendMessageAsync(topic, messages, partition: partition);
+        }
+
+        public Task<List<ProduceResponse>> SendMessageAsync(string topic, params Message[] messages)
+        {
+            return SendMessageAsync(topic, messages,acks:1);
+        }
+
         /// <summary>
         /// Get the metadata about a given topic.
         /// </summary>
         /// <param name="topic">The name of the topic to get metadata for.</param>
         /// <returns>Topic with metadata information.</returns>
-        public Topic GetTopic(string topic)
+        public Topic GetTopicFromCache(string topic)
         {
-            return _metadataQueries.GetTopic(topic);
+            return _metadataQueries.GetTopicFromCache(topic);
```

Comment:
we don't use it but In my opinion:
it need to work with ProtocolGateway.
we need to delete it form the Producer it's not his responsibility to check the offset( his only responsibility it's to Producer  message).

it not need to be part of brokerRouter but it need 3 Main API

Consumer
Producer
Offset

so i don't think we want to refactor it yet